lib/iris/_concatenate.py

```diff
@@ -992,6 +993,12 @@ def register(
             match = self._sequence(
                 coord_signature.dim_extents[dim_ind], candidate_axis
             )
+            if error_on_mismatch and not match:
+                msg = ["Dimensional coordinate extents differ"]
```

Comment:
This isn't as descriptive as the awrning message, something like `f"Found cubes with overlap on concatenate axis {candidate_axis}."` as below might be better. Also, this message shouldn't have to be in a list.